tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15119,7 +15179,28 @@ static bool nodeInputIsCompatible(StringRef &typeName,
       .Default(false);
 }
 
-void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr) {
+void DiagnoseComputeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr,
```

Comment:
I don't think we should be passing the shader attribute down to these functions.  Just pass the `ShaderKind` enum value decoded from the attribute in the calling function.